From 7546f821f7ab8c903d118717d476ba2b1cb056b3 Mon Sep 17 00:00:00 2001
From: "Tobias C. Berner" <tcberner@gmail.com>
Date: Sun, 20 Nov 2016 12:16:30 +0100
Subject: [PATCH] Fix build on FreeBSD-i386.

/wrkdirs/usr/ports/x11/kf5-kdelibs4support/work/kdelibs4support-5.27.0/src/kdeui/kpixmapcache.cpp:1161:88:
 error: non-constant-expression cannot be narrowed from type 'quint32' (aka 'unsigned int')
 to 'time_t' (aka 'int') in initializer list [-Wc++11-narrowing]
    KPixmapCacheIndexHeader indexHeader = { {0}, KPIXMAPCACHE_VERSION, 0, d->mCacheId, d->mTimestamp };
                                             ^~~~~~~~~~~~~
/wrkdirs/usr/ports/x11/kf5-kdelibs4support/work/kdelibs4support-5.27.0/src/kdeui/kpixmapcache.cpp:1161:88:
 note: override this message by inserting an explicit cast
    KPixmapCacheIndexHeader indexHeader = { {0}, KPIXMAPCACHE_VERSION, 0, d->mCacheId, d->mTimestamp };
                                             ^~~~~~~~~~~~~
                                             static_cast<time_t>( )

https://git.reviewboard.kde.org/r/129441

---
 src/kdeui/kpixmapcache.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kdeui/kpixmapcache.cpp b/src/kdeui/kpixmapcache.cpp
index 2d17571..b7a688b 100644
--- src/kdeui/kpixmapcache.cpp
+++ src/kdeui/kpixmapcache.cpp
@@ -1158,7 +1158,7 @@ bool KPixmapCache::recreateCacheFiles()
 
     // We can't know the full size until custom headers written.
     // mmapFiles() will take care of correcting the size.
-    KPixmapCacheIndexHeader indexHeader = { {0}, KPIXMAPCACHE_VERSION, 0, d->mCacheId, d->mTimestamp };
+    KPixmapCacheIndexHeader indexHeader = { {static_cast<time_t>(0)}, KPIXMAPCACHE_VERSION, 0, d->mCacheId, static_cast<time_t>(d->mTimestamp) };
     memcpy(indexHeader.magic, KPC_MAGIC, sizeof(indexHeader.magic));
 
     indexfile.write(reinterpret_cast<char *>(&indexHeader), sizeof indexHeader);
-- 
2.10.1

